.github/PULL_REQUEST_TEMPLATE.md

```diff
@@ -0,0 +1,20 @@
+# Description
+
+Please provide a summary of your changes and the reason the changes are being made.
+
+## Type of change
+
+Please delete options that are not relevant.
+
+  - [ ] Bug fix (non-breaking change which fixes an issue)
+  - [ ] New feature (non-breaking change which adds functionality)
+  - [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
+  - [ ] This change requires a documentation update
+
+# Checklist:
+
+  - [ ] My code follows the style guidelines of this project
+  - [ ] I have performed a self-review of my code
+  - [ ] I have commented my code, particularly in hard-to-understand areas
+  - [ ] I have made corresponding changes to the documentation
+  - [ ] My changes generate no new warnings
```

Comment:
More context here? What warnings?

Reply:
Added an example - generic but focuses more on static analysis and usage.